src/data/chats/mod.rs

```diff
@@ -75,22 +118,60 @@ impl Chats {
         }
     }
 
+    pub fn get_current_chat_model_file<'a>(&'a self, available_model_files: &'a [File]) -> &File {
+        let file_id = self.get_current_chat().unwrap().borrow().file_id.clone();
+        let file = available_model_files
+            .iter()
+            .find(|file| file.id == file_id)
+            .expect("Attempted to start chat with a no longer existing file");
+
+        file
+    }
+
+    pub fn set_current_chat(
```

Comment:
Ah ok.. Just to wrap up my previous comments here is a proposal to, I think, simplify things:

This function should be:

```rust
    pub fn set_current_chat(&mut self, chat_id: ChatID, file: &File) -> Result<()> {
        self.current_chat_id = Some(chat_id);

        if self
            .loaded_model_id
            .as_ref()
            .map_or(true, |m| *m != file.id)
        {
            self.load_model(file)?;
        }

        Ok(())
    }
```

And in `Store` we should have:

```rust
    pub fn set_current_chat(&mut self, chat_id: ChatID) {
        let available_files: Vec<File> = self
            .downloads
            .downloaded_files
            .iter()
            .map(|d| d.file.clone())
            .collect();

        let file_id = self.chats.get_current_chat().unwrap().borrow().file_id.clone();
        let file = available_files
            .iter()
            .find(|file| file.id == file_id)
            .expect("Attempted to start chat with a no longer existing file");

        if self
            .chats
            .set_current_chat(chat_id, &file)
            .is_ok()
        {
            // TODO: This breaks, FIX THE PREFERENCES!!!
             ....
            // self.preferences
            //     .set_current_chat_model(current_chat_model_file_id);
        }
    }
```

The main goal is to avoid passing around the `available_model_files` because there is no need for it.

Here is a complete diff for this idea: https://gist.github.com/jmbejar/0a94ae5c823f375bff081f45e9af65c2

@fmzbl what do you think?

Reply:
sure, this makes sense, I probably used Chats as a bigger "implementator" and didnt let store handle things, this way we can use the store more as the composer, so nice, it is probably better

Reply:
yea looking at the code again makes sense to do it in the store, tough its a bit confusing that chats is public and sometime we interface with it and sometimes we dont from the outside, so having two public (tough one accessed through the chats field) set_current_chat that do diffrent things can be a bit confussing, but for now i feel like its better ye

Reply:
The name duplication is a fair pain point. Renaming `Store.set_current_chat` to `Store.select_chat`, which is a name that makes sense from the caller point of view.